File: 896-monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        boolean increase = true, decrease = true;
        for (int i = 1; i < nums.length; i++) {
            if (!(nums[i - 1] >= nums[i]))
                increase = false;
            if (!(nums[i - 1] <= nums[i]))
                decrease = false;
        }
        return increase || decrease;
    }
}

File: 896-monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        mincr , mdecr = True, True
        for i in range(1,len(nums)):
            if not (nums[i-1] <= nums[i]):
                mincr = False
            if not (nums[i-1] >= nums[i]):
                mdecr = False
        return mincr or mdecr
